CMakeLists.txt

```diff
@@ -206,6 +206,7 @@ option(WITH_MEMORY_ANALYZER
 add_subdirectory(src)
 add_subdirectory(regression)
 add_subdirectory(unit)
+add_subdirectory(cpp_api)
```

Comment:
cpp_api, cprover_api, cprover_api_cpp, cbmc_api, api-... The naming is very inconsistent. We need to settle on a single name that is used throughout. I'd say let's go with `cprover-api-cpp`.

Reply:
The `cpp_api` is the directory name, which I've left for now because while I'm playing around with the rust part, I was thinking of renaming the folder into `api/cpp` and move the rust bit into `api/rust`.

I guess this also refers to the class name, which I can look into naming more appropriately.